zilliqa/src/consensus.rs

```diff
@@ -566,8 +582,11 @@ impl Consensus {
             Ok(()) => {}
             Err(e) => {
                 if let Some(e) = e.downcast_ref::<MissingBlockError>() {
-                    warn!("missing finalized block1");
-                    info!(?e, "missing block when checking block proposal");
+                    info!(?e, "missing block when checking block proposal - try and request the parent from the network: {}", block.header.number.saturating_sub(1));
+
+                    let random_peer = self.get_random_other_peer();
+                    self.block_store
+                        .request_blocks(random_peer, block.header.number.saturating_sub(1))?;
```

Comment:
Would there be any harm in requesting up to the proposal head? Given network latency, if it happens that the proposal passes before the request is processed, then you'll be a block behind still and might have to repeat this process, whereas if you request it then you automatically catch up.

If it doesn't, then worst case the response will just have one less block than expected which I believe block_store handles without issue.